airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

```diff
@@ -118,11 +126,16 @@ def truncate_identifier_name(self, input_name: str, custom_limit: int = -1) -> s
             prefix = input_name[: limit - middle - 1]
             suffix = input_name[1 - middle :]
             # Add extra characters '__', signaling a truncate in identifier
-            print(f"Truncating {input_name} (#{len(input_name)}) to {prefix}__{suffix} (#{2 + len(prefix) + len(suffix)})")
-            input_name = f"{prefix}__{suffix}"
+            print(f"Truncating {input_name} (#{len(input_name)}) to {prefix}_{suffix} (#{2 + len(prefix) + len(suffix)})")
+            mid = "__"
+            if conflict:
+                mid = f"_{conflict_solver}"
+            input_name = f"{prefix}{mid}{suffix}"
 
         return input_name
 
+    # conflict_stream_name____conflict_stream_name_ab3 => conflict_stream_name___flict_stream_name_ab3
+    # conflict_stream_name_conflict_stream_name_ab3    => conflict_stream_name___flict_stream_name_ab3
```

Comment:
this is probably better illustrated in the unit tests instead?

Reply:
removed from code, I was using to debugg the problem with mysql circle dependency in dbt